cli/code_scanner.py

```diff
@@ -367,38 +390,49 @@ def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, s
     return scan_result
 
 
-def perform_scan_async(cycode_client, zipped_documents: InMemoryZip, scan_type: str,
+def perform_scan_async(context: click.Context, cycode_client, zipped_documents: InMemoryZip, scan_type: str,
                        scan_parameters: dict) -> ZippedFileScanResult:
     scan_async_result = cycode_client.zipped_file_scan_async(zipped_documents, scan_type, scan_parameters)
     logger.debug("scan request has been triggered successfully, scan id: %s", scan_async_result.scan_id)
 
-    return poll_scan_results(cycode_client, scan_async_result.scan_id)
+    return poll_scan_results(context, cycode_client, scan_async_result.scan_id)
 
 
-def perform_commit_range_scan_async(cycode_client, from_commit_zipped_documents: InMemoryZip,
+def perform_commit_range_scan_async(context: click.Context, cycode_client, from_commit_zipped_documents: InMemoryZip,
                                     to_commit_zipped_documents: InMemoryZip, scan_type: str,
                                     scan_parameters: dict, timeout: int = None) -> ZippedFileScanResult:
     scan_async_result = \
         cycode_client.multiple_zipped_file_scan_async(from_commit_zipped_documents, to_commit_zipped_documents,
                                                       scan_type, scan_parameters)
     logger.debug("scan request has been triggered successfully, scan id: %s", scan_async_result.scan_id)
-    return poll_scan_results(cycode_client, scan_async_result.scan_id, timeout)
+    return poll_scan_results(context, cycode_client, scan_async_result.scan_id, timeout)
 
 
-def poll_scan_results(cycode_client, scan_id: str, polling_timeout: int = None):
+def poll_scan_results(context: click.Context, cycode_client, scan_id: str, polling_timeout: int = None):
     if polling_timeout is None:
         polling_timeout = configuration_manager.get_scan_polling_timeout_in_seconds()
 
+    printer = ResultsPrinter()
+    last_scan_update_at = None
+    output_type = context.obj['output']
     end_polling_time = time.time() + polling_timeout
+    spinner = Halo(spinner='dots')
+    spinner.start("Scan in progress")
     while time.time() < end_polling_time:
         logger.debug("scan in progress")
         scan_details = cycode_client.get_scan_details(scan_id)
+        if scan_details.scan_update_at is not None and scan_details.scan_update_at != last_scan_update_at:
+            last_scan_update_at = scan_details.scan_update_at
```

Comment:
updated_at

Reply:
It's how it's called from DB:
<img width="446" alt="image" src="https://user-images.githubusercontent.com/104304449/223155589-2d47768c-da0e-4a6b-9ecf-f4c5b0a1eb79.png">